test/functional/p2p_private_broadcast.py

```diff
@@ -0,0 +1,312 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017-present The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""
+Test how locally submitted transactions are sent to the network when private broadcast is used.
+
+The topology in the test is:
+
+Bitcoin Core (tx_originator)
+               ^   v       The default full-outbound + block-only connections
+               |   |       (MAX_OUTBOUND_FULL_RELAY_CONNECTIONS + MAX_BLOCK_RELAY_ONLY_CONNECTIONS):
+               |   |
+               |   *-----> SOCKS5 Proxy ---> P2PInterface (self.socks5_server.conf.destinations[0]["node"])
+               |   |
+               |   *-----> SOCKS5 Proxy ---> P2PInterface (self.socks5_server.conf.destinations[1]["node"])
+               |   ...
+               |   |       The private broadcast TX recipients
+               |   |       (NUM_PRIVATE_BROADCAST_PER_TX)
+               |   |       plus maybe some feeler or extra block only connections:
+               |   |
+               |   *-----> SOCKS5 Proxy ---> Bitcoin Core (node1) (self.socks5_server.conf.destinations[i]["node"] is None)
+               |   |                          ^
+               |   |                          *----< P2PInterface (far_observer) (to check Bitcoin Core relays the tx)
+               |   |
+               |   *-----> SOCKS5 Proxy ---> P2PInterface (self.socks5_server.conf.destinations[i + 1]["node"])
+               |   ...
+               |
+               *---------< P2PDataStore (observer_inbound)
+"""
+
+from test_framework.p2p import (
+    P2PDataStore,
+    P2PInterface,
+    P2P_SERVICES,
+    P2P_VERSION,
+)
+from test_framework.messages import (
+    CAddress,
+    CInv,
+    MSG_WTX,
+    msg_inv,
+)
+from test_framework.socks5 import (
+    Socks5Configuration,
+    Socks5Server,
+)
+from test_framework.test_framework import (
+    BitcoinTestFramework,
+)
+from test_framework.util import (
+    MAX_NODES,
+    assert_equal,
+    p2p_port,
+)
+from test_framework.wallet import (
+    MiniWallet,
+)
+
+MAX_OUTBOUND_FULL_RELAY_CONNECTIONS = 8
+MAX_BLOCK_RELAY_ONLY_CONNECTIONS = 2
+NUM_INITIAL_CONNECTIONS = MAX_OUTBOUND_FULL_RELAY_CONNECTIONS + MAX_BLOCK_RELAY_ONLY_CONNECTIONS
+NUM_PRIVATE_BROADCAST_PER_TX = 5
+
+# Fill addrman with these addresses. Must have enough Tor addresses, so that even
+# if all 10 default connections are opened to a Tor address (!?) there must be more
+# for private broadcast.
+addresses = [
```

Comment:
In 9297437af2c2f45a7b9a12be78fa0e597c54ac79: Since the addrman is deterministic now, you can remove addresses that are conflicting.

Reply:
Also, maybe we do not need this verification anymore?
```py
if res["success"]:
    self.log.info(f"Added {addr} to tx_originator's addrman")
else:
    self.log.info(f"Could not add {addr} to tx_originator's addrman (collision?)")
```

Reply:
Right, removed the conflicting address and reduced the logging to log only if adding to the addrman fails.